src/pages/RestaurantDetail.tsx

```diff
@@ -0,0 +1,96 @@
+//import { useTranslation } from "react-i18next";
+import {
+  Box,
+  Stack,
+  Title,
+  AreaLoader,
+  Tiles,
+  List,
+} from "@buildo/bento-design-system";
+import { useGetRestaurantDetails } from "../hooks";
+import { Feedback } from "@buildo/bento-design-system";
+import { useEffect } from "react";
+interface RestID {
+  id: string;
+}
+
+function RestaurantDetail({ id }: RestID) {
+  const { isLoading, isError, data, refetch } = useGetRestaurantDetails(id);
+
+  useEffect(() => {
+    refetch();
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
+
+  if (isLoading) {
+    return <AreaLoader message="Loading..."></AreaLoader>;
+  }
+
+  if (isError) {
+    return (
+      <Feedback
+        size="medium"
+        title="Error!"
+        description="Something went wrong!"
+        status="negative"
+      />
+    );
+  }
+
+  if (id == "0" || isError) {
+    return (
+      <Feedback
+        size="medium"
+        title="Error!"
+        description="Something went wrong!"
+        status="negative"
+      />
+    );
+  }
```

Comment:
The second case makes the first redundant, but the second one should not be needed if the handling of the restaurant ID is refactored as suggested above.